openmc/cmfd.py

```diff
@@ -0,0 +1,3072 @@
+                                  'file {}'.format(filename))
+            else:
+                # Overwrite CMFD values from statepoint
+                if (openmc.lib.master() and
+                        openmc.lib.settings.verbosity >= 5):
+                    print(' Loading CMFD data from {}...'.format(filename))
+                    sys.stdout.flush()
+                cmfd_group = f['cmfd']
+                self._cmfd_on = cmfd_group.attrs['cmfd_on']
+                self._feedback = cmfd_group.attrs['feedback']
+                self._feedback_begin = cmfd_group.attrs['feedback_begin']
+                self._tally_begin = cmfd_group.attrs['tally_begin']
+                self._time_cmfd = cmfd_group.attrs['time_cmfd']
+                self._time_cmfdbuild = cmfd_group.attrs['time_cmfdbuild']
+                self._time_cmfdsolve = cmfd_group.attrs['time_cmfdsolve']
+                self._window_size = cmfd_group.attrs['window_size']
+                self._window_type = cmfd_group.attrs['window_type']
+                self._k_cmfd = list(cmfd_group['k_cmfd'])
+                self._dom = list(cmfd_group['dom'])
+                self._src_cmp = list(cmfd_group['src_cmp'])
+                self._balance = list(cmfd_group['balance'])
+                self._entropy = list(cmfd_group['entropy'])
+                self._reset = list(cmfd_group['reset'])
+                self._albedo = cmfd_group['albedo'][()]
+                self._coremap = cmfd_group['coremap'][()]
+                self._egrid = cmfd_group['egrid'][()]
+                self._indices = cmfd_group['indices'][()]
+                self._current_rate = cmfd_group['current_rate'][()]
+                self._flux_rate = cmfd_group['flux_rate'][()]
+                self._nfiss_rate = cmfd_group['nfiss_rate'][()]
+                self._openmc_src_rate = cmfd_group['openmc_src_rate'][()]
+                self._p1scatt_rate = cmfd_group['p1scatt_rate'][()]
+                self._scatt_rate = cmfd_group['scatt_rate'][()]
+                self._total_rate = cmfd_group['total_rate'][()]
+
+                # Overwrite CMFD mesh properties
+                cmfd_mesh_name = 'mesh ' + str(cmfd_group.attrs['mesh_id'])
+                cmfd_mesh = f['tallies']['meshes'][cmfd_mesh_name]
+                self._mesh.dimension = cmfd_mesh['dimension'][()]
+                self._mesh.lower_left = cmfd_mesh['lower_left'][()]
+                self._mesh.upper_right = cmfd_mesh['upper_right'][()]
+                self._mesh.width = cmfd_mesh['width'][()]
+
+                # Store tally ids from statepoint run
+                sp_tally_ids = list(cmfd_group['tally_ids'])
+
+        # Set CMFD variables not in statepoint file
+        default_egrid = np.array([_ENERGY_MIN_NEUTRON, _ENERGY_MAX_NEUTRON])
+        self._energy_filters = not np.array_equal(self._egrid, default_egrid)
+        self._n_resets = len(self._reset)
+        self._mat_dim = np.max(self._coremap) + 1
```

Comment:
Optional: Would help readability to clarify the `+ 1` offset here with a comment.  My guess would be, to produce a value that is categorically higher than the `max(...)` on anything in the CMFD coremap dataset?